backend/store/principal.go

```diff
@@ -215,7 +216,25 @@ func listUserImpl(ctx context.Context, txn *sql.Tx, find *FindUserMessage) ([]*U
 		where, args = append(where, fmt.Sprintf("principal.deleted = $%d", len(args)+1)), append(args, false)
 	}
 
-	query := `
+	var with, join string
+	if v := find.ProjectID; v != nil {
+		with = `WITH all_members AS (
```

Comment:
I believe it's not the best solution, maybe @RainbowDashy can help optimize it later.

Reply:
😌👌